File: Stack/Stack/strCal.cs
using System;
using System.Collections.Generic;

public class Calculator
{
    public static int Calculate(string s)
    {
        s += "+"; // additional last op
        List<int> stack = new List<int>();

        int curNum = 0;
        char prevOp = '+';
        foreach (char c in s)
        {
            if (char.IsDigit(c))
            {
                curNum = curNum * 10 + int.Parse(c.ToString());
            }
            else if (c == ' ')
            {
                continue;
            }
            else // ops
            {
                if (prevOp == '+')
                {
                    stack.Add(curNum);
                }
                else if (prevOp == '-')
                {
                    stack.Add(-curNum);
                }
                else if (prevOp == '*')
                {
                    stack[stack.Count - 1] *= curNum;
                }
                else if (prevOp == '/')
                {
                    stack[stack.Count - 1] = (int)(stack[stack.Count - 1] / curNum);
                }

                curNum = 0;
                prevOp = c;
            }
        }

        int result = 0;
        foreach (int num in stack)
        {
            result += num;
        }

        return result;
    }

    public static void Main(string[] args)
    {
        int result = Calculate("7 - 6 / 3 + 3 * 2 + 4");
        Console.WriteLine(result);
    }
}

File: Stack/strCal.py
def calculate(s: str) -> int:
    s+='+' #additional last op
    stack = []
    
    cur_num = 0
    prev_op = '+'
    for c in s:
      if c.isdigit():
        cur_num = cur_num*10 + int(c)
        
      elif c == ' ':
        continue     
              
      else: #ops
        if prev_op == '+':
          stack.append(cur_num)

        elif prev_op == '-':
          stack.append(-cur_num)
          
        elif prev_op == '*':
          stack[-1] = stack[-1]*cur_num      

        elif prev_op == '/':
          stack[-1] = int(stack[-1]/cur_num)

        cur_num = 0
        prev_op = c
        
    return sum(stack)

calculate(s='7 - 6 / 3 + 3 * 2 + 4')
